src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -63,27 +57,38 @@ public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackage
             var nuGetProject = _solutionManager.Value.GetNuGetProject(projectSafeName);
             try
             {
+                _solutionManager.Value.SaveProject(nuGetProject);
```

Comment:
because in case the migration fails we might end up losing other changes that the user might have made.